### ISSUE TITLE
NMOS HMOS 6502 accuracy

### ISSUE THREAD

**Issue by miniupnp** [redcode/6502#2]
ASL, INC and possibly other read/write instructions do two writes on an original 6502.
They first write the original value, then the modified one.
The CMOS 65C02 behave differently.

That can cause issues when working on memory mapped I/O, such as what is done on the C64.

**Comment by miniupnp** [comment]
I'm not sure about the Ricoh 2A03/2A07 of the NES

**Comment by redcode** [comment]
Thanks, I've made several changes on the emulator since last version, I'll upload them to a new branch soon. I'm checking all instructions, memory access and cycles too.

**Comment by miniupnp** [comment]
regarding the differences between the different 6502 versions, do you plan to add different modes for the emulator ?

**Comment by redcode** [comment]
> regarding the differences between the different 6502 versions, do you plan to add different modes for the emulator ?

Yes, of course. I'm working on my NES emu, and I want to use this core for other machines too. I'll add support for the undocumented opcodes and for other 6502 models.

**Comment by redcode** [comment]
I've uploaded my most recent code to the branch WIP. I'll release a new version of this core in January 2020.
